20 спринт

David-Arcanone opened the pull request.

David-Arcanone committed:
задание спринта 20
корутины

David-Arcanone committed:
задание спринта 20
корутины

IliaAnK commented on app/src/main/java/com/practicum/playlistmaker/player/ui/PlayerViewModel.kt:
```diff
@@ -1,76 +1,49 @@
 class PlayerViewModel(private val myPlayerInteractor: PlayerInteractor) : ViewModel() {
-    private val mainThreadHandler = Handler(Looper.getMainLooper())
+    private var timerUpdateJob: Job? = null
+    private var myMediaPlayer: MediaPlayer = MediaPlayer()
```
Обязательно:
`MediaPlayer` это своего рода источник данных. А согласно требованиям 15 и 16 спринтов с источниками данных должен работать репозиторий, а не `ViewModel`. Нужно выделить работу с `MediaPlayer` в `Repository` и не забыть про `domain` слой через который `ViewModel` будет обращаться к нему.

David-Arcanone replied:
По поводу вашего замечания "На экране «Аудиоплеер» BottomNavBar и разделитель должены скрываться.": 
В 19 спринте нет слайда с плеером в фигме, как в светлой так и в темных темах.
https://www.figma.com/design/mDybm1hLkL6hXmMoPmR5Au/Playlist-Maker_Tab-Bar-(YP)?node-id=1-1066&node-type=canvas&t=8kWftC6fc0BjyIeo-0
https://www.figma.com/design/mDybm1hLkL6hXmMoPmR5Au/Playlist-Maker_Tab-Bar-(YP)?node-id=0-1&node-type=canvas&t=ZCF4RD4CNB2119ph-0
Ничего по поводу видимости панели навигации в окне проигрывателя там не написано, только по поводу кнопки назад.
А в 20 спринте вообще нет ссылки на фигму, откуда вы ссылаетесь на то, к чему у меня доступа нет?
Чтоб в следующих потоках не повторяли эту ошибку, добавьте картинку с Плеером в фигму.

IliaAnK replied:
Могу сослаться на [макеты](https://www.figma.com/design/pQ7xipgQ42cS4Kee0Vqo0y/Playlist-Maker.-Basic-state-player-(YP)?node-id=1-455&node-type=frame&t=KP4s6S2LhHU98kyA-0) 9 спринта.

Поскольку в предыдущем спринте не было требований по изменению поведения на экране "Аудиоплеер", то и  `BottomNavBar`'а там быть не должно

David-Arcanone committed:
задание спринта 20
-изменил верстку согласно замечанию о навигационном меню -перенес медиаПлеер обратно в репозиторий.
